lib/restful_resource/http_client.rb

```diff
@@ -265,6 +271,7 @@ def http_request(request)
       when 429 then raise HttpClient::TooManyRequests.new(request, response)
       when 502 then raise HttpClient::BadGateway.new(request, response)
       when 503 then raise HttpClient::ServiceUnavailable.new(request, response)
+      when 504 then raise HttpClient::GatewayTimeout.new(request, response)
```

Comment:
are we sure we want to muffle it everywhere?

Reply:
Yes, it already raises a generic error, this will make it more specific.